kedro/framework/cli/starters.py

```diff
@@ -91,19 +129,16 @@ def new(
             "Cannot use the --directory flag without a --starter value."
         )
 
-    # The `astro-iris` was renamed to `astro-airflow-iris`, but old (external) documentation
-    # and tutorials still refer to `astro-iris`. The below line checks if a user has entered old
-    # `astro-iris` as the starter name and changes it to `astro-airflow-iris`.
-    starter_name = (
-        "astro-airflow-iris" if starter_name == "astro-iris" else starter_name
-    )
-    if starter_name in _STARTER_ALIASES:
+    starter_aliases = _get_starters_aliases()
+
+    if starter_name in starter_aliases.keys():
         if directory:
             raise KedroCliError(
                 "Cannot use the --directory flag with a --starter alias."
             )
-        template_path = _STARTERS_REPO
-        directory = starter_name
+        template_path = starter_aliases[starter_name]["repo"]
+        # "directory" is an optional key for starters from plugins, so if the key is not present we will use "None".
+        directory = starter_aliases[starter_name].get("directory")
         checkout = checkout or version
```

Comment:
I think we should be able to specify `checkout` in `_STARTER_ALIASES` also. The order of precedence should be:
1. command line `checkout` argument
2. `checkout` given in `_STARTER_ALIASES`
3. `version`

Reply:
I've done it, even if I am not sure we should let checkout be part of the config (see below)

Reply:
As discussed below, I completely remove the "checkout" option of this PR.